nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java

```diff
@@ -12,105 +12,91 @@
  */
 package org.sonatype.repository.helm;
 
-import java.util.Collections;
+import java.util.EnumMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+
+import org.sonatype.nexus.common.collect.NestedAttributesMap;
+import org.sonatype.repository.helm.internal.database.HelmProperties;
 
 /**
- * Object for storing Helm specific attributes from Chart.yaml
- *
- * @since 0.0.1
+ * @since 1.0.next
  */
-public final class HelmAttributes
+public class HelmAttributes
 {
-  private String description;
-  private String engine;
-  private String home;
-  private String icon;
-  private String appVersion;
-  private List<String> keywords;
-  private List<Map<String, String>> maintainers;
-  private String name;
-  private List<String> sources;
-  private String version;
-
-  public String getDescription() {
-    return description;
-  }
-
-  public void setDescription(final String description) {
-    this.description = description;
-  }
+  private final Map<HelmProperties, Object> attributesEnumMap;
 
-  public String getEngine() {
-    return engine;
+  public HelmAttributes() {
+    attributesEnumMap = new EnumMap<>(HelmProperties.class);
   }
 
-  public void setEngine(final String engine) {
-    this.engine = engine;
+  public HelmAttributes(final Map<String, Object> attributesMap) {
+    attributesEnumMap = new EnumMap<>(HelmProperties.class);
+    attributesMap.forEach((key, value) -> {
+      Optional<HelmProperties> propertyOpt = HelmProperties.findByPropertyName(key);
+      if (value != null && propertyOpt.isPresent()) {
+        attributesEnumMap.put(propertyOpt.get(), value);
```

Comment:
HelmProperties.findByPropertyName always return optional so in this place, we do not need to check for null
You can replace your code
`    HelmProperties.findByPropertyName(key)
          .ifPresent(properties -> attributesEnumMap.put(properties, value));`

Reply:
value which we get from file could be null